.github/workflows/schedule-posts.txt

```diff
@@ -0,0 +1,36 @@
+name: Publish posts scheduled for today
+
+on:
+  schedule:
+    # Run every day at 23:30 UTC or manually run
+    - cron: "30 23 * * *"
+  workflow_dispatch:
+
+jobs:
+  publish_drafts:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout main branch
+        uses: actions/checkout@v4
+        with:
+          ref: master
+
+      - name: Check for drafts and move to posts
+        run: |
+          today=$(date +'%Y-%m-%d')
+          echo "Today is $today"
+          shopt -s nullglob
+          for file in _drafts/${today}-*.md; do
+            echo "Found draft: $file"
+            mv "$file" "_posts/"
+            echo "Moved $file to _posts/"
+          done
+
+      - name: Commit and push changes
+        run: |
+          git config user.name "github-actions"
+          git config user.email "actions@github.com"
+          git add _posts/
+          git add _drafts/
+          git commit -m "Posted Scheduled Drafts on $TODAY" || echo "No changes to commit"
```

Comment:
Since you removed the exporting of `TODAY` to GITHUB_ENV, I think this will not print correctly (not sure). My suggestion is to add the export as it was before, but in the same `Check for drafts and move to posts` and always use `TODAY`.